src/components/page-configuration/change-repository-form.tsx

```diff
@@ -1,60 +1,91 @@
 import * as PropTypes from 'prop-types';
-import { useEffect, useState } from 'react';
-
-import imageDeployKey from './deploy-key.png';
-import useSaveRepository from './use-save-repository';
-import imageWebhook from './webhook02.png';
+import { useEffect, useState, Validator } from 'react';
+import { connect } from 'react-redux';
 
 import { Alert } from '../alert';
-import { Code } from '../code';
-import { ApplicationRegistrationModelValidationMap } from '../../models/application-registration';
 import { RequestState } from '../../state/state-utils/request-states';
-import { configVariables } from '../../utils/config';
+import {
+  ApplicationRegistrationModel,
+  ApplicationRegistrationModelValidationMap,
+} from '../../models/application-registration';
+import imageDeployKey from '../configure-application-github/deploy-key02.png';
+import imageWebhook from '../configure-application-github/webhook01.png';
+import { Code } from '../code';
 import { copyToClipboard } from '../../utils/string';
+import { configVariables } from '../../utils/config';
+import { useSaveRepository } from './use-save-repository';
 
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const radixZoneDNS = configVariables.RADIX_CLUSTER_BASE;
-const webhookURL = `https://webhook.${radixZoneDNS}/events/github`;
 
-export const ChangeRepositoryForm = (props) => {
-  const app = props.app;
-  const [savedRepository, setSavedRepository] = useState(props.repository);
-  const [repository, setRepository] = useState(props.repository);
-  const [saveState, saveFunc, resetState] = useSaveRepository(props.appName);
-  const [previousRepository, setPreviousRepository] = useState();
+export interface ChangeRepositoryFormProps {
+  appName: string;
+  repository: string;
+  acknowledgeWarnings?: boolean;
+  app?: ApplicationRegistrationModel;
+}
+
+export const ChangeRepositoryForm = (
+  props: ChangeRepositoryFormProps
+): JSX.Element => {
+  const [currentRepository, setCurrentRepository] = useState(props.repository);
+  const [editedRepository, setEditedRepository] = useState(props.repository);
+  const [useAcknowledgeWarnings, setAcknowledgeWarnings] = useState(false);
+  const [modifyState, saveFunc, resetState] = useSaveRepository(props.appName);
+  const applicationRegistration =
+    modifyState.data?.applicationRegistration ?? undefined;
+  const operationWarnings = modifyState.data?.warnings ?? undefined;
   const [updateRepositoryProgress, setUpdateRepositoryProgress] =
     useState(false);
+  const webhookURL = `https://webhook.${radixZoneDNS}/events/github?appName=${props.appName}`;
 
   useEffect(() => {
-    setRepository(props.repository);
-    setUpdateRepositoryProgress(false);
-    return () => setPreviousRepository(repository);
+    setEditedRepository(currentRepository);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [props.repository]);
+  }, [currentRepository]);
 
   const handleSubmit = (ev) => {
```

Comment:
add typing (remember to update imports `FormEvent from 'react'`)
```suggestion
  const handleSubmit = (ev: FormEvent<HTMLFormElement>): void => {
```